Sources/PubNubChatComponents/Message List/MessageListViewController.swift

```diff
@@ -133,4 +209,162 @@ open class ChatViewController: CollectionViewComponent {
+      messageCell: gestureRecognizedCell!,
+      messageReactionsComponentSize: preferredPickerViewSize
+    )
+  }
+  
+  @objc(animationControllerForDismissedController:)
+  public func animationController(forDismissed dismissed: UIViewController) -> UIViewControllerAnimatedTransitioning? {
+    return AddMessageReaction.AnimationController(duration: 0.15, isBeingPresented: false)
+  }
+  
+  @objc(animationControllerForPresentedController:presentingController:sourceController:)
+  public func animationController(forPresented presented: UIViewController, presenting: UIViewController, source: UIViewController) -> UIViewControllerAnimatedTransitioning? {
+    return AddMessageReaction.AnimationController(duration: 0.25, isBeingPresented: true)
+  }
+}
+
+// MARK: Showing message reactions
+
+private enum AddMessageReaction {
+  
+  // MARK: Custom presentation controller (UIPresentationController)
+
+  @objc
+  class PresentationController: UIPresentationController {
+    private let messageCell: MessageListItemCell?
+    private let messageReactionsComponentSize: CGSize
+    
+    private lazy var dimmingView: UIView = {
+      let view = UIView()
+      
+      view.frame = containerView?.bounds ?? .zero
+      view.backgroundColor = .clear
+      view.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(onTapGestureRecognized(_:))))
+      
+      return view
+    }()
+    
+    init(
+      presentedViewController: UIViewController,
+      presentingViewController: UIViewController?,
+      messageCell: MessageListItemCell,
+      messageReactionsComponentSize: CGSize
+    ) {
+      self.messageCell = messageCell
+      self.messageReactionsComponentSize = messageReactionsComponentSize
+      
+      super.init(
+        presentedViewController: presentedViewController,
+        presenting: presentingViewController
+      )
+    }
+    
+    override var frameOfPresentedViewInContainerView: CGRect {
+      
+      guard let containerView = containerView, let messageCell = messageCell else {
+        return super.frameOfPresentedViewInContainerView
+      }
+      
+      let messageCellFrame = messageCell.superview?.convert(messageCell.frame, to: containerView) ?? .zero
+      
+      // Presents the reaction components beyond the cell that was pressed.
+      // If it doesn't fit at the screen, then the picker component is presented at the bottom of a view in which a presentation occurs
+      let x = messageCellFrame.minX
+      let y = min(containerView.frame.maxY - messageReactionsComponentSize.height - containerView.safeAreaInsets.bottom, messageCellFrame.maxY + 10)
+      
+      return CGRect(
+        x: x,
+        y: y,
+        width: messageReactionsComponentSize.width,
+        height: messageReactionsComponentSize.height
+      )
+    }
+    
+    @objc
+    func onTapGestureRecognized(_ sender: UIGestureRecognizer) {
+      presentedViewController.dismiss(animated: true)
```

Comment:
it is not coherent when after method declaration is empty new line

Reply:
Where's an empty new line? I cannot find it

Reply:
In some other methods, for example line 102, or computed property above. I mean it is not clear for me if we should have here empty line or not :)